File: 12-03-2025_Min_Cost_Climbing_Stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;
        
        if (n == 1) {
            return cost[0];
        }
        
        int prev2 = cost[0];
        int prev1 = cost[1];
        
        for (int i = 2; i < n; i++) {
            int curr = cost[i] + Math.min(prev1, prev2);
            prev2 = prev1;
            prev1 = curr;
        }
        
        return Math.min(prev1, prev2);
    }
}

File: 12-03-2025_Min_Cost_Climbing_Stairs.py
class Solution:
    def minCostClimbingStairs(self, cost):
        n = len(cost)
        
        if n == 1:
            return cost[0]
        
        prev2 = cost[0]
        prev1 = cost[1]
        
        for i in range(2, n):
            curr = cost[i] + min(prev1, prev2)
            prev2 = prev1
            prev1 = curr
        
        return min(prev1, prev2)
